app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/SupportService.scala

```diff
@@ -59,8 +59,9 @@ class SupportService @Inject() (
     (
       for {
         flow       <- ET.liftF(fetchSupportFlow(sessionId))
-        api        <- ET.fromEitherF(apmConnector.fetchExtendedApiDefinition(apiChoice))
-        updatedFlow = flow.copy(api = Some(SupportApi(apiChoice, api.name)))
+        apiName    <- if (apiChoice.value == "api-not-in-list") ET.liftF(Future.successful(""))
+                      else ET.fromEitherF(apmConnector.fetchExtendedApiDefinition(apiChoice).map(_.map(_.name)))
```

Comment:
And probably ET.fromEitherT(...).map(_.name) on the outside to save map map.

Reply:
Done.